src/core/bpf-cgroup-program-spec-parsers.c

```diff
@@ -62,6 +63,69 @@ static int create_bpf_cgroup_program_spec(
         return 0;
 }
 
+static int get_json_string_value(JsonVariant *json, const char* key, const char **value) {
+        JsonVariant *var;
+
+        var = json_variant_by_key(json, key);
+        if (!var)
+                return -ENOENT;
+
+        *value = json_variant_string(var);
+        if (!*value)
+                return -EINVAL;
+
+        return 0;
+}
+
+static int create_bpf_cgroup_program_spec_from_json(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *lvalue,
+                const char* rvalue,
+                Unit *u,
+                struct BpfCgroupProgramSpec **spec) {
+        enum bpf_attach_type attach_type;
+        const char *tmp, *bpffs_path;
+        uint32_t attach_flags = 0;
+        int rc;
+
+        if (isempty(rvalue))
+                return -ENOENT;
```

Comment:
please make sure to always log about all errors or about none on each function. right now this function logs for some errors but not for others.